src/main.cpp

```diff
@@ -34,36 +47,74 @@
 #        include "clientdlg.h"
 #    endif
 #endif
-#include "settings.h"
+
+#include "rpcserver.h"
+#include "serverrpc.h"
 #ifndef SERVER_ONLY
+#    include "clientrpc.h"
 #    include "testbench.h"
 #endif
-#include "util.h"
+
 #ifdef ANDROID
 #    include <QtAndroidExtras/QtAndroid>
 #endif
+
 #if defined( Q_OS_MACX )
 #    include "mac/activity.h"
 extern void qt_set_sequence_auto_mnemonic ( bool bEnable );
 #endif
-#include <memory>
-#include "rpcserver.h"
-#include "serverrpc.h"
-#ifndef SERVER_ONLY
-#    include "clientrpc.h"
-#endif
+
+// Global App Name *************************************************************
+
+static QString strAppName = APP_NAME; // Will be appended by " - ClientName" or "Server - ServerName" if those names are defined
+
+const QString GetAppName() { return strAppName; }
+
+// Helpers *********************************************************************
+
+const QString getServerNameFromInfo ( const QString strServerInfo )
+{
+    if ( !strServerInfo.isEmpty() )
+    {
+        QStringList servInfoParams = strServerInfo.split ( ";" );
+        if ( servInfoParams.count() > 0 )
+        {
+            return servInfoParams[0];
+        }
+    }
+
+    return QString();
+}
 
 // Implementation **************************************************************
 
 int main ( int argc, char** argv )
 {
+    int exit_code = 0;
```

Comment:
Give a comment here why we need the variable. (I still don't like having an exit code variable here...)

Reply:
Hmmm Why not ? Its the return value of main, and it's quite common practice to declare the returned value as first in a function.

Reply:
> Its the return value of main, and it's quite common practice to declare the returned value as first in a function.

But I don't see any other places in Jamulus where functions have this?

Reply:
> But I don't see any other places in Jamulus where functions have this?

Perhaps you have to look better than ;=))

Just do a search on `return ret` , `return res` or `return result` and you will already find a bunch of them...
Or search for `return _id;` since also all qt_metacall moc_ functions use this coding pattern!
Or take a look look at:

`CChannel::SetSockBufNumFrames`
`CChannel::PutAudioData`
`CChannel::GetData`
`CProtocol::GetValFromStream`
`CNetBuf::Get`
`CNetBufWithStats::Get`
`CBuffer::GetAvailData()`
`CClient::GetAndResetbJitterBufferOKFlag()`
`CServer::CreateChannelList()`
`CServer::FindChannel`
`CServer::PutAudioData`
`CServer::CreateLevelsForAllConChannels`
`CSettings::GetIniSetting`
`CSettings::GetNumericIniSet`
`CSettings::GetFlagIniSet`
...
All the same pattern, just different variable names....

This really is a standard coding practice....